src/metabase/driver/common/parameters/values.clj

```diff
@@ -121,22 +125,28 @@
                    :id field-id)
                  (throw (ex-info (str (deferred-tru "Can''t find field with ID: {0}" field-id))
                                  {:field-id field-id, :type qp.error-type/invalid-parameter}))))
-    :value (if-let [value-info-or-infos (or
-                                         ;; look in the sequence of params we were passed to see if there's anything
-                                         ;; that matches
-                                         (param-with-target params [:dimension [:template-tag (:name tag)]])
-                                         ;; if not, check and see if we have a default param
-                                         (default-value-for-field-filter tag))]
-             ;; `value-info` will look something like after we remove `:target` which is not needed after this point
-             ;;
-             ;;    {:type   :date/single
-             ;;     :value  #t "2019-09-20T19:52:00.000-07:00"}
-             ;;
-             ;; (or it will be a vector of these maps for multiple values)
-             (cond
-               (map? value-info-or-infos)        (dissoc value-info-or-infos :target)
-               (sequential? value-info-or-infos) (mapv #(dissoc % :target) value-info-or-infos))
-             i/no-value)}))
+    :value (or (when-let [value-info-or-infos (or
+                                               ;; look in the sequence of params we were passed to see if there's anything
+                                               ;; that matches
+                                               (param-with-target (map defaulted-param params) [:dimension [:template-tag (:name tag)]])
+                                               ;; if not, check and see if we have a default param
+                                               (default-value-for-field-filter tag))]
+                 ;; `value-info` will look something like after we remove `:target` which is not needed after this point
+                 ;;
+                 ;;    {:type   :date/single
+                 ;;     :value  #t "2019-09-20T19:52:00.000-07:00"}
+                 ;;
+                 ;; (or it will be a vector of these maps for multiple values)
+                 (let [has-value?    (some-fn :value :default)
+                       dissoc-target #(dissoc % :target)]
+                   (cond
+                     (map? value-info-or-infos)
+                     (when (has-value? value-info-or-infos)
```

Comment:
do we know why a `map?` ends up here but it fails `has-value?`? I'm wondering if there's another part that is doing something incorrect or if this is the proper place to put it. What's in these maps here if it doesn't have a value?

Reply:
This is the map I'm getting:
```
{:name "Date Filter",
  :slug "date_filter",
  :id "97ff7f0",
  :type :date/all-options,
  :default "past30days",
  :target [:dimension [:template-tag "created_at"]]}
  ```
  
  And you can also get the equivalent map with no `:default` if the default filter hasn't been set:
  
  
![image](https://user-images.githubusercontent.com/784417/112407334-e86f3880-8ce3-11eb-86cb-b1e13059f2fa.png)

Reply:
And if a `value-info` (or multiple of them) can get here without a value we might have a problem. Suppose there were a few of them but one did not have a default or value. The `every?` would fail and we would substitute a single `i/no-value` where there previously were three.

This code seems to assume that if there's a `value-info-or-value-infos` its either a well-formed map or a well-formed sequence. It seems like we should `(merge {:default i/no-value} value-info)` across the single or the collection here.

Reply:
I don't know too much but it seems proper to "backfill" these maps without a default to have a default of `i/no-value`? Then this code pathway will be much simpler i think

Reply:
Ah sorry i'm getting confused in the overloaded meaning of value. I'm confusing things more than helping sorry.